typescript/web/src/components/labeling-tool/options-tool-bar/ai-assistant-toolbar.tsx

```diff
@@ -0,0 +1,338 @@
+  const { runAiAssistant, loading } = useAiAssistant();
+  return (
+    <Button
+      data-testid="ai-assistant-annotate-button"
+      leftIcon={{ name: useAnnotateIcon(), fontSize: "1.3em" }}
+      onClick={runAiAssistant}
+      isLoading={loading}
+      loadingText="Running..."
+      tooltip={`Run AI Assistant [${keymap.runAiAssistant.key}]`}
+      px={2}
+    >
+      Annotate
+    </Button>
+  );
+};
+
+type ShowOptionsButtonProps = {
+  showOptions: boolean;
+};
+
+const ShowOptionsButton = ({ showOptions }: ShowOptionsButtonProps) => {
+  const { loading } = useAiAssistant();
+  return (
+    <MenuButton
+      as={Button}
+      tooltip={{ label: "Show options", placement: "bottom" }}
+      isActive={showOptions}
+      disabled={loading}
+      p={1}
+    >
+      <Icon name="more" />
+    </MenuButton>
+  );
+};
+
+const OptionsMenuList = () => {
+  const { useAutoPolygon, toggleUseAutoPolygon } = useAiAssistant();
+  return (
+    <MenuList>
+      <MenuItemOption isChecked={useAutoPolygon} onClick={toggleUseAutoPolygon}>
+        Post-process with Auto-Polygon
+        <BetaBadge />
+      </MenuItemOption>
+    </MenuList>
+  );
+};
+
+type AnnotateBodyProps = { showOptions: boolean };
+
+const AnnotateBody = ({ showOptions }: AnnotateBodyProps) => {
+  const {
+    aiAssistant: { labelType },
+    loading,
+  } = useAiAssistant();
+  const hasOptions = labelType === LabelType.Box;
+  return (
+    <ButtonGroup isAttached colorScheme="brand" disabled={loading}>
+      <AnnotateButton />
+      {hasOptions && <ShowOptionsButton showOptions={showOptions} />}
+    </ButtonGroup>
+  );
+};
+
+export type AnnotateProps = Pick<MenuProps, "isOpen">;
+
+export const AiAssistantAnnotateButtonGroup = (props: AnnotateProps) => (
+  <Menu {...props}>
+    {({ isOpen }) => (
+      <>
+        <AnnotateBody showOptions={isOpen} />
+        <OptionsMenuList />
+      </>
+    )}
+  </Menu>
+);
+
+const AiAssistantItem = ({ name, iconUrl }: AiAssistant) => (
+  <HStack>
+    <Image src={iconUrl} h={4} w={4} />
+    <Text>{name}</Text>
+  </HStack>
+);
+
+const AiAssistantListItem = (aiAssistant: AiAssistant) => {
+  const { id, summary } = aiAssistant;
+  return (
+    <Flex direction="column" data-testid={`ai-assistant-item-${id}`}>
+      <AiAssistantItem {...aiAssistant} />
+      <Text fontSize="xs">{summary}</Text>
+    </Flex>
+  );
+};
+
+export type AiAssistantComboBoxProps = Pick<
+  ComboBoxProps<AiAssistant, "id">,
+  "isOpen"
+>;
+
+export const AiAssistantComboBox = (props: AiAssistantComboBoxProps) => {
+  const { aiAssistant, setAiAssistant, loading } = useAiAssistant();
+  return (
+    <ComboBox
+      data-testid="ai-assistant-combobox"
+      {...props}
+      items={AI_ASSISTANTS}
+      compareProp="name"
+      item={AiAssistantItem}
+      listItem={AiAssistantListItem}
+      selectedItem={aiAssistant}
+      onChange={(value) => {
+        if (isNil(value)) return;
+        setAiAssistant(value);
+      }}
```

Comment:
Early returns are used in many places in our codebase and are considered to be a good practice (it was even noted on our onboarding presentation 😉 ) since they make the code take less lines, avoids extra indentation levels and makes the code more readable in complex cases.

Reply:
I could have written : 
```suggestion
      onChange={(value) => {
        if (!isNil(value)) setAiAssistant(value);
      }}
```